scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -2039,3 +2040,123 @@ def test_compare_as_davenport_as_euler():
             eul = rot.as_euler(seq)
             dav = rot.as_davenport(ax, order)
             assert_allclose(eul, dav, rtol=1e-12)
+
+
+def test_zero_length_rotation_construction():
+    r = Rotation.random(num=0)
+    assert len(r) == 0
+
+    r_ide = Rotation.identity(num=0)
+    assert len(r_ide) == 0
+
+    r_get = Rotation.random(num=3)[[]]
+    assert len(r_get) == 0
```

Comment:
Add a similar test with `[False, False, False]` indexing.

Reply:
As @scottshambaugh this might go into a separate test for indexing.

Reply:
I added an additional boolean-indexing test function in https://github.com/scipy/scipy/commit/6a76337fe3c74fbc86c57c628c695db3c359ff05

Replied above, too.